extensions/src/common/extension/decorators/newBlocks.ts

```diff
@@ -107,12 +36,14 @@ namespace Utility {
 namespace Argument {
     type TRemoveUtil<T extends any[]> = T extends [...infer R extends any[], BlockUtilityWithID] ? R : T;
     // Maya note: thought ToArguments was the equivalent, but TypeScript does not like it....
+    // Parker note: ^interesting! Also, if we keep this implementation, we'll need to handle InlineImages
     export type MapToScratch<T extends any[], Internal extends TRemoveUtil<T> = TRemoveUtil<T>> = {
         [k in keyof Internal]: Argument<Internal[k]>
     }
 }
 
-interface TemplateEngine<TBlockType> {
+// TODO: Restrict return based on Scratch type
```

Comment:
I actually mean that we should restrict the return based on the mapping we have of Block types to which method return types. `ReturnTypeByBlockType` I believe